### ISSUE TITLE
Key repeat cancelled/blocked by previous key's release event

### ISSUE THREAD

**Issue by marcomayer** [wez/wezterm#2412]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

wezterm 20220814-222723-f2ee48bd

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

It's not so easy to describe but the same bug happened in Alacritty (https://github.com/alacritty/alacritty/issues/2788), the solution there was to upgrade the winit dependency.

I notice that a lot as a vi-user, let's say I press the "h"-key and keep on pressing/repeating the key so a lot of "hhhhhhhh" are printed in the console. Now if I quickly switch to the "l"-key while stopping to hold the "h"-key, the "l" won't get triggered.

### To Reproduce

I'm on the latest macbook pro using the latest macos version. My macos keyboard settings are set to fastest repeat and lowest delay on repeat. Just fire up wezterm without any config and try it out.

### Configuration

no config

### Expected Behavior

Fast switching between keys should be non-blocking (as in Alacritty/Kitty/iterm2 for example). Would love to see this solved, it's the last thing holding me back from using wezterm as my daily driver.

**Comment by bew** [comment]
> Now if I quickly switch to the "l"-key while stopping to hold the "h"-key, the "l" won't get triggered.

So no `l` are sent at all? Or `l` are sent but they stop when your release `h` ?

**Comment by marcomayer** [comment]
> > Now if I quickly switch to the "l"-key while stopping to hold the "h"-key, the "l" won't get triggered.
> 
> So no `l` are sent at all? Or `l` are sent but they stop when your release `h` ?

One "l" is sent and then it blocks. Sorry wasn't clear on that in my description.

**Comment by wez** [comment]
Can you try setting `use_ime = false`?

**Comment by chmanie** [comment]
I don't mean to hijack this but I think I have exactly the same problem. Pressing two keys in fast succession won't trigger the key repeat of the last key pressed. I'm running wezterm under Linux/wayland and tried to set `use_ime` to `false` to no effect.

**Comment by wez** [comment]
@chmanie please open a separate issue; keyboard related stuff is highly platform dependent so this issue is almost certainly not the same as what you're seeing even though it sounds similar
